dist/bundle.js

```diff
@@ -710174,31 +710174,27 @@ function getHandlerContract(web3) {
 exports.getHandlerContract = getHandlerContract;
 function checkContentType(url) {
     return new Promise((resolve, reject) => {
-        let returnVal = { valid: false };
-        try {
-            fetch(url, { method: 'HEAD' })
-                .then(response => {
-                if (!response.ok) {
-                    returnVal.valid = false;
-                }
-                else if (response.status === 200) {
-                    const contentType = response.headers.get('content-type');
-                    let extension = getFileExtensionFromMimeType(contentType);
-                    returnVal.valid = true;
-                    returnVal.contentType = contentType;
-                    returnVal.extension = extension;
-                    returnVal.embed = !isValidDirect(extension);
-                    console.log('Content-Type:', contentType);
-                }
-                resolve(returnVal);
-            })
-                .catch(error => {
-                console.error('Error while fetching URL:', error);
-                resolve(returnVal);
-            });
-        }
-        catch (error) {
-        }
```

Comment:
Removed `try{}catch{}` and instead returned `resolved(returnVal)` where `returnVal.valid` is set to `false`.

Added tests to verify we return without throwing an exception when the URL is invalid.